### PR TITLE
PROFIN-23 | agrego endpoint de consulta por code

### PR DESCRIPTION
**Pull request by MarcosPereyra**

### CODE REVIEW

**Comment by code-race-2021** [comment]
# [Codecov](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/8?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) Report
> Merging [#8](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/8?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) (843a766) into [master](https://codecov.io/gh/code-race-2021/final-project-cruds/commit/30b939a19d36e942507d6a94e90943fdcffcb3fe?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) (30b939a) will **decrease** coverage by `2.73%`.
> The diff coverage is `0.00%`.

[Impacted file tree graph](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/8?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None)

```diff
@@             Coverage Diff              @@
##             master       #8      +/-   ##
============================================
- Coverage     95.79%   93.06%   -2.74%     
  Complexity      126      126              
============================================
  Files            20       20              
  Lines           238      245       +7     
  Branches          3        3              
============================================
  Hits            228      228              
- Misses            4       11       +7     
  Partials          6        6              
```


| [Impacted Files](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/8?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) | Coverage Δ | |
|---|---|---|
| [...va/com/coderace/controller/DeliveryController.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/8/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2UvY29udHJvbGxlci9EZWxpdmVyeUNvbnRyb2xsZXIuamF2YQ==) | `66.66% <0.00%> (-33.34%)` | :arrow_down: |
| [...ain/java/com/coderace/service/DeliveryService.java](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/8/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-c3JjL21haW4vamF2YS9jb20vY29kZXJhY2Uvc2VydmljZS9EZWxpdmVyeVNlcnZpY2UuamF2YQ==) | `84.61% <0.00%> (-15.39%)` | :arrow_down: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/8?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/8?src=pr&el=footer&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None). Last update [30b939a...843a766](https://codecov.io/gh/code-race-2021/final-project-cruds/pull/8?src=pr&el=lastupdated&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).

**Comment by MarcosPereyra** [comment]
Nei, ya agregué los tests de getByCode()

File: src/main/java/com/coderace/service/DeliveryService.java [hunk 1]
```diff
@@ -59,4 +61,11 @@ protected DeliveryResponseDTO buildDeliveryResponseDTO(Delivery delivery) {
 
         return responseDTO;
     }
+
+    public DeliveryResponseDTO getByCode(String code) {
+        return this.repository.getByCode(code)
+                .map(this::buildDeliveryResponseDTO)
+                .orElseThrow(() -> new BadRequestException(HttpStatus.NOT_FOUND.value(),
+                        String.format("Product with code [%s] not found", code)));
```
Comment (NeiMeli) [inline src/main/java/com/coderace/service/DeliveryService.java]: Menor, cambiaría el mensaje para que diga Delivery en vez de Product. Y lo mismo en el test

Comment (MarcosPereyra) [reply src/main/java/com/coderace/service/DeliveryService.java]: Corregido, Neii
https://github.com/code-race-2021/final-project-cruds/pull/8/commits/6a862b094f4f8be33c279271595aafcf6803dd32

**Review by NeiMeli — APPROVED** [review]

**Pull request merged**